main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield i, index
+      index + 1
+    end
+  end
```

Comment:
# my_each_with_index
- [ ] `my_each_with_index` returns an enumerator if no block is given

     Failure/Error: expect(array.my_each_with_index).to be_an(Enumerator)
       expected nil to be a kind of Enumerator
##

- [ ] `my_each_with_index` when a block is given when self is an array returns the array itself after calling the given block once for each element in self

     Failure/Error: expect(array.my_each_with_index(&block)).to eq(array.each_with_index(&block))

       expected: [4, 2, 6, 0, 2, 7, 5, 3, 3, 1, 2, 1, 3, 8, 8, 1, 2, 7, 5, 3, 7, 8, 1, 3, 1, 5, 4, 0, 4, 8, 2, 7, 0, 0..., 8, 1, 7, 3, 5, 5, 6, 7, 5, 7, 4, 0, 2, 6, 2, 2, 7, 8, 6, 6, 4, 1, 1, 7, 7, 4, 1, 1, 5, 7, 4, 2, 3]
            got: 0

##

- [ ] `my_each_with_index` when a block is given when self is an array behaves like #each_with_index calls the given block once for each element in self

     Failure/Error: expect(my_each_with_index_output).to eq(each_with_index_output)

       expected: "Num: 5, idx: 0\nNum: 5, idx: 1\nNum: 5, idx: 2\nNum: 1, idx: 3\nNum: 1, idx: 4\nNum: 3, idx: 5\nNum:...: 4, idx: 94\nNum: 5, idx: 95\nNum: 4, idx: 96\nNum: 5, idx: 97\nNum: 8, idx: 98\nNum: 1, idx: 99\n"
            got: "Num: 0, idx: #<Enumerator:0x0000559ab87450d8>\nNum: 1, idx: #<Enumerator:0x0000559ab8744f48>\nNum: 2...>\nNum: 98, idx: #<Enumerator:0x0000559ab8773280>\nNum: 99, idx: #<Enumerator:0x0000559ab8773118>\n"

##

- [ ] `my_each_with_index` when a block is given when self is a range returns the range itself after calling the given block once for each element in self

     Failure/Error: expect(range.my_each_with_index(&block)).to eq(range.each_with_index(&block))

     NameError:
       undefined local variable or method `length' for 5..50:Range
##

- [ ] `my_each_with_index` when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self

     Failure/Error: enum.my_each_with_index(&block)

     NameError:
       undefined local variable or method `length' for 5..50:Range

##

- [ ] `my_each_with_index` when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self

     Failure/Error: enum.my_each_with_index(&block)

     ArgumentError:
       wrong number of arguments (given 0, expected 1)
##